GCC 15 intentionally warns for AVX10.1 aliases generated by -march=native.
Expect only that diagnostic when an independent native CPU probe finds
the corresponding feature; preserve all options and run assertions.
See https://gcc.gnu.org/gcc-15/changes.html

--- a/gcc/testsuite/g++.dg/pr90773-1d.C
+++ b/gcc/testsuite/g++.dg/pr90773-1d.C
@@ -1,3 +1,4 @@
+/* { dg-warning "'-mavx10\\.1' is aliased to 512 bit since GCC14\\.3 and GCC15\\.1 while '-mavx10\\.1-256' and '-mavx10\\.1-512' will be deprecated in GCC 16 due to all machines 512 bit vector size supported(?=\\r|\\n|$)" "native AVX10.1 alias" { target crossforge_native_avx10_1 } 0 } */
 // { dg-do run }
 // { dg-options "-O2" }
 // { dg-additional-options "-march=native" { target { i?86-*-* x86_64-*-* } } }
